### PR TITLE
remove checks for metachain nodes in sovereign

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
## Reasoning behind the pull request
- 
- 
- 
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: node/trieIterators/factory/delegatedListHandlerFactory.go [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
-	"github.com/multiversx/mx-chain-go/node/trieIterators/disabled"
 )
 
 // CreateDelegatedListHandler will create a new instance of DirectStakedListHandler
 func CreateDelegatedListHandler(args trieIterators.ArgTrieIteratorProcessor) (external.DelegatedListHandler, error) {
 	//TODO add unit tests
-	if args.ShardID != core.MetachainShardId {
-		return disabled.NewDisabledDelegatedListProcessor(), nil
-	}
+	//TODO2: only if the chain type is sovereign, then allow shard nodes no respond to this endpoint
```
Comment (sasurobert) [inline node/trieIterators/factory/delegatedListHandlerFactory.go:11]: resolve this.

File: node/trieIterators/factory/stakeValuesHandlerFactory.go [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
-	"github.com/multiversx/mx-chain-go/node/trieIterators/disabled"
 )
 
 // CreateTotalStakedValueHandler will create a new instance of TotalStakedValueHandler
 func CreateTotalStakedValueHandler(args trieIterators.ArgTrieIteratorProcessor) (external.TotalStakedValueHandler, error) {
-	if args.ShardID != core.MetachainShardId {
-		return disabled.NewDisabledStakeValuesProcessor(), nil
-	}
+	// TODO: based on chain type, allow shards nodes as well
+	//if args.ShardID != core.MetachainShardId {
```
Comment (sasurobert) [inline node/trieIterators/factory/stakeValuesHandlerFactory.go:11]: resolve.

File: process/block/preprocess/sovereignChainTransactions.go [hunk 3]
```diff
@@ -53,7 +53,9 @@ func (sct *sovereignChainTransactions) CreateAndProcessMiniBlocks(haveTime func(
 	//TODO: Check through "allin system tests" if this value of 2x (normal txs + scheduled txs) is enough or could be raised,
 	//as with the new mechanism of proposing and parallel processing for proposer and validators, we are not limited anymore
 	//for a subround block duration of maximum 25% from the whole round. Actually the processing time could / will go up to 60% - 80%.
-	gasBandwidth := sct.economicsFee.MaxGasLimitPerBlock(sct.shardCoordinator.SelfId()) * 2.0
+
+	// TODO: maybe re-add the 2.0 multiplier
+	gasBandwidth := sct.economicsFee.MaxGasLimitPerBlock(sct.shardCoordinator.SelfId())
```
Comment (sasurobert) [inline process/block/preprocess/sovereignChainTransactions.go:58]: it sovereign has its own economics.toml then no need for 2x.

**Comment by mariusmihaic** [comment]
This was already refactored and implemented.

**Pull request closed without merging**